client-hc/src/main/java/com/graphhopper/api/GraphHopperWeb.java

```diff
@@ -258,13 +258,26 @@ private Request createPostRequest(GHRequest ghRequest) {
         requestJson.put("elevation", ghRequest.getHints().getBool("elevation", elevation));
         requestJson.put("optimize", ghRequest.getHints().get("optimize", optimize));
 
-        Map<String, String> hintsMap = ghRequest.getHints().toMap();
-        for (String hintKey : hintsMap.keySet()) {
+        Map<String, Object> hintsMap = ghRequest.getHints().toMap();
+        for (Map.Entry<String, Object> entry : hintsMap.entrySet()) {
+            String hintKey = entry.getKey();
             if (ignoreSetForPost.contains(hintKey))
                 continue;
 
-            String hint = hintsMap.get(hintKey);
-            requestJson.put(hintKey, hint);
+            // try proper JSON conversion at least for numbers and booleans
+            Object hint = entry.getValue();
+            if (hint instanceof Boolean)
+                requestJson.put(hintKey, (Boolean) hint);
+            else if (hint instanceof Integer)
+                requestJson.put(hintKey, (Integer) hint);
+            else if (hint instanceof Long)
+                requestJson.put(hintKey, (Long) hint);
+            else if (hint instanceof Float)
+                requestJson.put(hintKey, (Float) hint);
+            else if (hint instanceof Double)
+                requestJson.put(hintKey, (Double) hint);
+            else
+                requestJson.put(hintKey, hint.toString());
```

Comment:
Similar to AbstractMatrixRequester: Can we keep the actual object instead of turning into a string here? But we can probably add this when we really need it. I just wonder in which case the `toString()` implementation will be of actual use (if it cannot be parsed later its useless or might even break the json format (?)).

Reply:
Ah, you mean with putPOJO ... will try.

Reply:
Yes, maybe, does this use Jackson internally (when it is finally serialized)?